cpp/cmake_modules/SetupCxxFlags.cmake

```
@@ -32,6 +32,8 @@ if(NOT DEFINED ARROW_CPU_FLAG)
     set(ARROW_CPU_FLAG "ppc")
   elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "s390x")
     set(ARROW_CPU_FLAG "s390x")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "riscv64")
+    set(ARROW_CPU_FLAG "riscv")
   else()
     set(ARROW_CPU_FLAG "x86")
```

We might want to be stricter in our detection of x86 targets, e.g. (untested):
```suggestion
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "AMD64|X86|x86|i[3456]86")
    set(ARROW_CPU_FLAG "x86")
  else()
    set(ARROW_CPU_FLAG "unknown")
```
(see https://stackoverflow.com/a/70498851 for possible values)

---

cc @cyb70289

---

`x86_64` should be tested too as that's what `uname -m` returns here.

---

"MATCHES" should match a substring (it does a regex match), though it's worth checking.